tests/utils.py

```diff
@@ -45,25 +49,80 @@ def print_success_msg() -> None:
 # ====================================================================
 
 
+class PackageDependencies(NamedTuple):
+    typeshed_pkgs: tuple[str, ...]
+    external_pkgs: tuple[str, ...]
+
+
 @cache
-def read_dependencies(distribution: str) -> tuple[str, ...]:
+def read_dependencies(distribution: str) -> PackageDependencies:
     with Path("stubs", distribution, "METADATA.toml").open("rb") as f:
         data = tomli.load(f)
-    requires = data.get("requires", [])
-    assert isinstance(requires, list)
-    dependencies = []
-    for dependency in requires:
+    dependencies = data.get("requires", [])
+    assert isinstance(dependencies, list)
+    typeshed, external = [], []
+    for dependency in dependencies:
         assert isinstance(dependency, str)
-        assert dependency.startswith("types-"), f"unrecognized dependency {dependency!r}"
-        dependencies.append(dependency[6:].split("<")[0])
-    return tuple(dependencies)
+        if dependency.startswith("types-"):
```

Comment:
I don't see a problem here, but there are two cases where this logic may do something slightly surprising / different from stub_uploader which will consider something a typeshed stub iff it's something it's uploaded. So maybe worth adding a comment:
> This function may consider things to be typeshed stubs even if they haven't been uploaded to PyPI. If a typeshed stub is removed, this function will consider it to be an external dependency.

Reply:
Your comment made me realise that there is also an edge case that this function doesn't currently handle: stubs which are uploaded with nonstandard names to PyPI. It'll break, for example, if we ever get a stubs package that depends on `types-pika-ts`.

This is unlikely to come up in real life, but I may as well fix it.

Reply:
How does 42f722f look?